### ISSUE TITLE
CI: generate testing docker images on demand when dependencies are updated

### ISSUE THREAD

**Issue by flavorjones** [sparklemotion/nokogiri#1998]
I've got some work in progress on hoe-gemspec to enable the use of external gemspecs within Hoe:

https://github.com/flavorjones/hoe-gemspec/issues/1

This would enable dependabot integration and automated updating of the gemspec ... which means we could then trigger updates to the default testing images used in the test pipelines (see #1863 for background context).

This is a placeholder for that work.

**Comment by flavorjones** [comment]
Blocked on https://github.com/flavorjones/hoe-gemspec/issues/1

**Comment by flavorjones** [comment]
Closing in favor of #2247.
